lms/djangoapps/lti_provider/models.py

```diff
@@ -20,30 +23,98 @@ class LtiConsumer(models.Model):
     specific settings, such as the OAuth key/secret pair and any LTI fields
     that must be persisted.
     """
-    consumer_name = models.CharField(max_length=255)
+    consumer_name = models.CharField(max_length=255, unique=True)
     consumer_key = models.CharField(max_length=32, unique=True, db_index=True)
     consumer_secret = models.CharField(max_length=32, unique=True)
+    instance_guid = models.CharField(max_length=255, null=True)
+
+    @staticmethod
+    def get_or_supplement(instance_guid, consumer_key):
+        """
+        The instance_guid is the best way to uniquely identify an LTI consumer.
+        However according to the LTI spec, the instance_guid field is optional
+        and so cannot be relied upon to be present.
+
+        This method first attempts to find an LtiConsumer by instance_guid.
+        Failing that, it tries to find a record with a matching consumer_key.
+        This can be the case if the LtiConsumer record was created as the result
+        of an LTI launch with no instance_guid.
+
+        If the instance_guid is now present, the LtiConsumer model will be
+        supplemented with the instance_guid, to more concretely identify the
+        consumer.
+
+        In practice, nearly all major LTI consumers provide an instance_guid, so
+        the fallback mechanism of matching by consumer key should be rarely
+        required.
+        """
+        consumer = None
+        if instance_guid:
+            try:
+                consumer = LtiConsumer.objects.get(instance_guid=instance_guid)
```

Comment:
It can be. It can't be blank. Uniqueness is enforced at the database level, and it can be nullable because NULL != NULL in SQL. Blanks can't be. In Django you almost always use blank for this sort of thing because that's Django's default practice for interpreting form data. Making something that uses nulls to represent empty means that some of the builtin form functionality won't work properly.

That being said, if you use `unique=True, null=True` and create an `LtiConsumer` explicitly setting `instance_guid` to `None`, then it will save.